### PR TITLE
Allow Running Overridden Blueprints

### PR DESCRIPTION
**Pull request by blimmer**

### CODE REVIEW

**Comment by jgwhite** [comment]
This is very much in line with what @trabus and I have been discussing. :+1:

I’d like to do a tiny bit of bikeshedding over the separator (with no strong preference):

``` sh
ember g package-name/blueprint-name
ember g package-name:blueprint-name
ember g blueprint-name # top-most (package|addon)’s blueprint wins
```

**Comment by stefanpenner** [comment]
> ember g package-name:blueprint-name

I like `:` or `::` or `@`

`ember g blueprint-name@package-name` is likely my favorite

**Comment by blimmer** [comment]
I'm open to any naming convention that fits well with existing Ember CLI conventions. 

The only one I don't particularly like is `/` because I think it's a bit confusing given that `/` means something else when generating nested routes.

**Comment by trabus** [comment]
To me, `ember g package-name:blueprint-name foo` seems the most natural, but I'm open to introducing `ember g blueprint-name@package-name foo` as a new convention, since `@` makes it like saying blueprint-name **_at**_ package-name.

**Comment by jgwhite** [comment]
Cool, I think we’re agreed on `:` as the separator. :+1: from me for this RFC.

File: active/0000-allow-running-overridden-blueprint.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+However, there are new features and tests that I'd like to start developing in JavaScript. As implemented today, since `ember-cli-coffeescript` overwrites the default ember-cli blueprints, I have to [flip a switch](https://github.com/kimroen/ember-cli-coffeescript/tree/v0.10.0#blueprints) every time in my configuration, which [short-circuits](https://github.com/kimroen/ember-cli-coffeescript/blob/v0.10.0/index.js#L26) the `blueprintsPath` in `ember-cli-coffeescript` and makes default ember-cli blueprints available (since the add-on is no longer overriding the blueprint definition) and will result in a generated JavaScript file.
+
+# Detailed design
+
+What I'm proposing is having an easier way to run a different add-on's blueprint from the command line. For instance, instead of flipping the switch in the configuration file, it'd be great if I could leave `ember-cli-coffeescript` add-on in tact and tell `ember generate` that I want to run the blueprint from the base `ember-cli` package, instead of `ember-cli-coffeescript`'s overridden version.
+
+I imagine this would be something like
+```
+ember generate ember-cli:route
```
Comment (stefanpenner) [inline active/0000-allow-running-overridden-blueprint.md:23]: `ember:route` I believe would make the most sense.

**Comment by stefanpenner** [comment]
:+1: for this, any implementors interested?

**Comment by stefanpenner** [comment]
Thoughts on

```
ember g model foo --blueprint-source=ember
ember g model foo --blueprint-source=<addon-name>
```

Some examples

running default blueprints provided by ember-cli would be:

```
ember g model foo --blueprint-source=ember-cli
```

running them from a specific node_module, (like ember-data).

```
ember g model foo --blueprint-source=ember-data
```

**Comment by blimmer** [comment]
@stefanpenner I'm into it. LGTM.

**Comment by stefanpenner** [comment]
@blimmer mind updating the RFC accordingly? Also, do you have any interest in implementing, or are you unavailable time wise.

**Comment by rwjblue** [comment]
We are working on closing the ember-cli/rfcs repo in favor of using a single central RFC's repo for everything. This was laid out in https://emberjs.github.io/rfcs/0300-rfc-process-update.html. 

Sorry for the troubles, but would you mind reviewing to see if this is still something we need, and if so migrating this over to emberjs/rfcs?

Thank you!

**Pull request closed without merging**